lib/net/peer.js

```diff
@@ -62,6 +62,7 @@ class Peer extends EventEmitter {
     this.socket = null;
     this.brontide = new BrontideStream();
     this.identityKey = this.options.identityKey;
```

Comment:
The options above are not actually receiving `PeerOptions` but `PoolOptions`. Passing `PeerOptions` will cause it to break in the cleartext case.

Reply:
Resolved by using `PeerOptions`